tiledb/bioimg/converters/png.py

```diff
@@ -0,0 +1,215 @@
+
+    @property
+    def channels(self) -> Sequence[str]:
+        if self.webp_format is WebpInputFormat.WEBP_RGB:
+            self._logger.debug(f"Webp format: {WebpInputFormat.WEBP_RGB}")
+            return "RED", "GREEN", "BLUE"
+        self._logger.debug(f"Webp format is not: {WebpInputFormat.WEBP_RGB}")
+        color_map = {"R": "RED", "G": "GREEN", "B": "BLUE"}
+        # Use list comprehension to convert the short form to full form
+        rgb_full = [color_map[color] for color in self._png.getbands()]
+        return rgb_full
+
+    @property
+    def level_count(self) -> int:
+        level_count = 1
+        self._logger.debug(f"Level count: {level_count}")
+        return level_count
+
+    def level_dtype(self, level: int = 0) -> np.dtype:
+        dtype = np.uint8
+        self._logger.debug(f"Level {level} dtype: {dtype}")
+        return dtype
+
+    def level_shape(self, level: int = 0) -> Union[Tuple[int, ...], Any]:
+        if level != 0:
+            raise ValueError(
+                "PNG images are not pyramidal and consist of only one level"
+            )
+        # Even after converting to RGB the size is not updated to 3d from 2d
+        l_shape = (*self._png.size, 3)
+        self._logger.debug(f"Level {level} shape: {l_shape}")
+        return l_shape
+
+    @property
+    def webp_format(self) -> WebpInputFormat:
+        self._logger.debug(f"Keyframe photometric: {self._png.mode}")
+        if self._png.mode == "RGB":
+            return WebpInputFormat.WEBP_RGB
+        return WebpInputFormat.WEBP_NONE
+
+    def level_image(
+        self, level: int = 0, tile: Optional[Tuple[slice, ...]] = None
+    ) -> np.ndarray:
+
+        if tile is None:
+            return np.asarray(self._png)
+        else:
+            return np.asarray(self._png)[tile]
+
+    def level_metadata(self, level: int) -> Dict[str, Any]:
+        # Common with group metadata since there are no multiple levels
+        writer_kwargs = dict(metadata=self._metadata)
+        return {"json_write_kwargs": json.dumps(writer_kwargs)}
+
+    @property
+    def group_metadata(self) -> Dict[str, Any]:
+        writer_kwargs = dict(metadata=self._metadata)
+        self._logger.debug(f"Group metadata: {writer_kwargs}")
+        return {"json_write_kwargs": json.dumps(writer_kwargs)}
+
+    @property
+    def image_metadata(self) -> Dict[str, Any]:
+        self._logger.debug(f"Image metadata: {self._metadata}")
+        color_generator = iter_color(np.dtype(np.uint8), len(self.channels))
+
+        channels = []
+        for idx, channel in enumerate(self.channels):
+            channel_metadata = {
+                "id": f"{idx}",
+                "name": f"Channel {idx}",
+                "color": (next(color_generator)),
+            }
+            channels.append(channel_metadata)
+        self._metadata["channels"] = channels
+        return self._metadata
+
+    @property
+    def original_metadata(self) -> Dict[str, Any]:
```

Comment:
It ssems like `group_metadata` is a generalization of this ...

* Maybe put the two next to each other
* Maybe have one call the other (no big deal, they're already short functions)
* Please add docstrings

Reply:
It happens in this specific Converter that the two coincide. It is indeed repeated information but we need to follow the design as it is. PNGs are not pyramidal whole slides so incorporating the converter into a design that was mostly built for multi-resolution imagery requires that. I can modify so one function calls the other.

Reply:
On doctstrings comment:  All these `converters/ readers/ writers` follow the base protocols `ImageReader`, `ImageWriter`, `ImageConverter` in `base.py` . This way they implement the protocol and the `docstrings` for each of these functions/properties resides there to avoid repetition and it is common for all implementations.